canonical-web-and-design/ubuntu.com
# Prefill name and address

⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️
### BEFORE MERGING / AFTER QA
Contract API url and Stripe public key must be updated to production values. They can be found [here](https://docs.google.com/document/d/1BG2mRWUF9pHAiwLPgm-7gelrZdstjlkQit84kXBum4A), and also here:
- contract API url: https://contracts.canonical.com
- Stripe public key: pk_live_68aXqowUeX574aGsVck8eiIE

⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️

## Done

- Added a call to `/v1/accounts/{account}/customer-info/stripe` to get the name and address from Stripe
- Refactored the way changing your payment methods fills the form to use the same function for both

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/advantage/subscribe
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- If logged out, log in
- if logged in, see that your address should be pre filled to the last one you used to make a purchase
- make a new purchase with a new address, refresh the page and see that it should be pre filled next time

## Issue / Card

Fixes #8757 

## Side note
I've noticed that in the response from `/v1/accounts/{account}/customer-info/stripe` we also have the paymentId and the 4 last digits of the card. Have we thought of pre-filling the card number input with something like 'Card ending in 1234 [Use another card button]'? 
That would turn the purchase process in a single click (granted there's no ATV number required)
